datatypes/src/error.rs

```diff
@@ -129,11 +127,86 @@ pub enum Error {
     },
 }
 
+#[derive(Debug)]
+pub struct ArrowWrappedError {
+    error: arrow::error::ArrowError,
+}
+
+impl AsRef<arrow::error::ArrowError> for ArrowWrappedError {
+    fn as_ref(&self) -> &ArrowError {
+        &self.error
+    }
+}
+
+impl std::fmt::Display for ArrowWrappedError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        self.error.fmt(f)
+    }
+}
+
+impl std::error::Error for ArrowWrappedError {}
+
+impl From<arrow::error::ArrowError> for ArrowWrappedError {
+    fn from(error: ArrowError) -> Self {
+        Self { error }
+    }
+}
+
+impl PartialEq for ArrowWrappedError {
+    fn eq(&self, other: &Self) -> bool {
+        match (self.as_ref(), other.as_ref()) {
+            (
+                arrow::error::ArrowError::MemoryError(e1),
+                arrow::error::ArrowError::MemoryError(e2),
+            )
+            | (
+                arrow::error::ArrowError::ParseError(e1),
+                arrow::error::ArrowError::ParseError(e2),
+            )
+            | (
+                arrow::error::ArrowError::SchemaError(e1),
+                arrow::error::ArrowError::SchemaError(e2),
+            )
+            | (
+                arrow::error::ArrowError::ComputeError(e1),
+                arrow::error::ArrowError::ComputeError(e2),
+            )
+            | (arrow::error::ArrowError::CsvError(e1), arrow::error::ArrowError::CsvError(e2))
+            | (arrow::error::ArrowError::JsonError(e1), arrow::error::ArrowError::JsonError(e2))
+            | (arrow::error::ArrowError::IoError(e1), arrow::error::ArrowError::IoError(e2))
+            | (
+                arrow::error::ArrowError::InvalidArgumentError(e1),
+                arrow::error::ArrowError::InvalidArgumentError(e2),
+            )
+            | (
+                arrow::error::ArrowError::ParquetError(e1),
+                arrow::error::ArrowError::ParquetError(e2),
+            ) => e1 == e2,
+            (arrow::error::ArrowError::DivideByZero, arrow::error::ArrowError::DivideByZero)
+            | (
+                arrow::error::ArrowError::DictionaryKeyOverflowError,
+                arrow::error::ArrowError::DictionaryKeyOverflowError,
+            ) => true,
+            (arrow::error::ArrowError::ExternalError(_), _)
```

Comment:
this fails if the error is external? why not do a catch all case for other?

Reply:
Unfortunately, since it supplies no check for equality.
If I catch all then I have to clue when they add another case in the future that requires a check.

Reply:
4fb5cb3 
I've removed the bound and as such this overhead. All the rest compiled fine, so I guess there is no need for `PartialEq` on error types.